### PR TITLE
Cherry-picks for foreman 2.2.1

### PR DESCRIPTION
**Pull request by tbrisker**
Changes include https://projects.theforeman.org/issues/29938 and a followup to https://projects.theforeman.org/issues/30535 fixing a typo

### CODE REVIEW

**Review by ekohl — APPROVED** [review]

**Review by wbclark — APPROVED** [review]
https://github.com/theforeman/puppet-foreman/pull/902/commits/cd1378e0603f30983c13fc6ed071316fcab941e2 is backwards incompatible because that logging layout was added in Foreman 2.2

however the purpose of these cherry picks seems to be to support Foreman 2.2.1 so that should be fine

normally a change marked as backwards incompatible should be accompanied by a bump in the major release, but in this case i could continue to list it as backwards incompatible in the changelog while only bumping the minor release

after merging the cherry picks I'll prepare a release PR

File: manifests/params.pp [hunk 1]
```diff
@@ -189,7 +189,7 @@
   # Application logging
   $logging_level = 'info'
   $logging_type = 'file'
-  $logging_layout = 'pattern'
+  $logging_layout = 'multiline_request_pattern'
```
Comment (ekohl) [inline manifests/params.pp:192]: Thinking more about it, it wouldn't be backwards incompatible if we kept this the same. There's a migration in the installer anyway so it isn't needed there.

Comment (tbrisker) [reply manifests/params.pp:192]: makes sense

Comment (wbclark) [reply manifests/params.pp:192]: done with https://github.com/theforeman/puppet-foreman/pull/904

**Pull request merged**